cmd/bucket-replication_test.go

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2015-2021 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	xhttp "github.com/minio/minio/cmd/http"
+
+	"github.com/minio/minio/pkg/bucket/replication"
+)
+
+var configs = []replication.Config{
+	{ // Config0 - Replication config has no filters, existing object replication enabled
+		Rules: []replication.Rule{
+			{
+				Status:                    replication.Enabled,
+				Priority:                  1,
+				DeleteMarkerReplication:   replication.DeleteMarkerReplication{Status: replication.Enabled},
+				DeleteReplication:         replication.DeleteReplication{Status: replication.Enabled},
+				Filter:                    replication.Filter{},
+				ExistingObjectReplication: replication.ExistingObjectReplication{Status: replication.Enabled},
+				SourceSelectionCriteria: replication.SourceSelectionCriteria{
+					ReplicaModifications: replication.ReplicaModifications{Status: replication.Enabled},
+				},
+			},
+		},
+	},
+}
+
+var replicationConfigTests = []struct {
+	info         ObjectInfo
```

Comment:
Don't use numbers, use names for reference:
```suggestion
	name         string
	info         ObjectInfo
```